### PR TITLE
Collector database model

### PR DESCRIPTION
**Pull request by pallavi2209**

### CODE REVIEW

File: db/model/collector.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      {
+        name: 'CollectorUniqueLowercaseName',
+        unique: true,
+        fields: [
+          seq.fn('lower', seq.col('name')),
+        ],
+      },
+    ],
+  });
+  return Collector;
```
Comment (iamigo) [inline db/model/collector.js]: needs writers, and assign creator as writer on create

**Comment by pallavi2209** [comment]
@iamigo @shriramshankar updated the PR as per comments and discussion.

File: db/model/collector.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      validate: { isEmail: true },
+    },
+    helpUrl: {
+      type: dataTypes.STRING(constants.fieldlen.url),
+      validate: { isUrl: true },
+    },
+    registered: {
+      type: dataTypes.BOOLEAN,
+      allowNull: false,
+      defaultValue: true,
```
Comment (shriramshankar) [inline db/model/collector.js]: If this is like isPublished, should the default be false instead?

File: db/model/collector.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      afterCreate(inst /* , opts*/) {
+        // Add createdBy user to Collector writers.
+        if (inst.createdBy) {
+          return new seq.Promise((resolve, reject) =>
+            inst.addWriter(inst.createdBy)
+            .then(() => resolve(inst))
+            .catch((err) => reject(err))
+          );
+        }
+
```
Comment (shriramshankar) [inline db/model/collector.js]: need a beforeDelete hook to set isDeleted.

File: db/model/collector.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        return inst;
+      }, // hooks.beforeCreate
+    }, // hooks
+    indexes: [
+      {
+        name: 'CollectorUniqueLowercaseName',
+        unique: true,
```
Comment (shriramshankar) [inline db/model/collector.js]: isDeleted should also be a part of the unique index.

File: db/model/collector.js (hunk 2 above)
Comment (pallavi2209) [inline db/model/collector.js]: Yeah, I thought so too, but I think this field corresponds to the 'collector is registered with Refocus' and if someone is creating a collector, then they want it registered by default. @iamigo What do you think?

Comment (iamigo) [inline db/model/collector.js]: This record will only ever be created by a collector registering itself, so registered should default to true.

**Pull request merged**